app/scripts/controllers/modals/setHomePageModal.js

```diff
@@ -0,0 +1,74 @@
+'use strict';
+/* jshint unused: false */
+
+/**
+ * @ngdoc function
+ * @name openshiftConsole.controller:SetHomePageModalController
+ * @description
+ * # SetHomePageModalController
+ * Controller of the openshiftConsole
+ */
+angular.module('openshiftConsole')
+  .controller('SetHomePageModalController', function ($scope, $uibModalInstance, HomePagePreference) {
+
+    $scope.initialPreferedHomePage = HomePagePreference.getHomePagePreference();
+    $scope.preferedHomePage = $scope.initialPreferedHomePage;
+
+    $scope.selectedProject = null;
+    $scope.noProjects = false;
+
+    $scope.preSelectedProjectName = HomePagePreference.getProjectOverviewPage();
+
+    $scope.onProjectSelected = function(project) {
+      $scope.selectedProject = project;
+    };
+
+    $scope.onOpen = function() {
+      $scope.preferedHomePage = "project-overview";
+    };
+
+    $scope.$on('no-projects-cannot-create', function() {
+      $scope.noProjects = true;
+    });
+
+    $scope.disableSaveButton = function() {
+      // if radiobuttons not changed
+      if ($scope.preferedHomePage === $scope.initialPreferedHomePage) {
+        if ($scope.preferedHomePage === 'project-overview') {
+          if (!$scope.selectedProject) {
```

Comment:
Can we just check `form.$pristine` instead to avoid all of this logic? The nested ifs are really hard to follow.